Backend/Hub/AuthHub.cs

```diff
@@ -69,21 +91,25 @@ public async Task<string> GetSessionToken()
             return (await Database.UserSessions.FromSignalRId(Context.UserIdentifier))?.ToToken();
         }
 
-        public async Task<bool> Register(string username, string password, string email)
+        public async Task<bool> Register(string username, string password, string email, double curLat, double curLong)
         {
-            if (await Database.Authenticators.AsAsyncEnumerable().AnyAsync(e => e.Email == email))
+            if (await Database.Authenticators.AsQueryable().AnyAsync(e => e.Email == email))
                 return false;
 
-            Authenticator auth = new(email, CreatePasswordHash(password), DateTime.UtcNow, new(0, username, email));
+            User user = new(0, username, email);
+            Authenticator auth = new(email, CreatePasswordHash(password), DateTime.UtcNow, user);
 
-            // TODO: add user to a new group, sync the group with users, and generate a new challenge. There should be an extenstion method for each one.
-            //Group Extension method to generate new random challenge (not in prev challenge), add current challenge if one exists to prev challenge list of group + all members
+            await auth.User.NewGroup(Database.Challenges, curLat, curLong);
 
-            await auth.User.NewGroup(Database.Challenges);
             await Groups.AddToGroupAsync(Context.UserIdentifier, auth.User.GroupMember.Group.Id.ToString());
-            await Database.Groups.AddAsync(auth.User.GroupMember.Group);
 
             await Database.Authenticators.AddAsync(auth);
+            await Database.Groups.AddAsync(auth.User.GroupMember.Group);
+
+            //Add to sessionlog
+            var entry = new SessionLogEntry(SessionLogEntryType.UserCreated, user.Id + "registered", DateTime.UtcNow, user);
```

Comment:
+ "registered" is unnecessary. Remove them everywhere you wrote this type of thing.